packages/oats-fast-check/src/generator.ts

```diff
@@ -67,7 +67,7 @@ export class GenType extends fc.Arbitrary<any> {
       assert.fail('todo intersection value generation');
     }
     if (type.type === 'named') {
-      return named(type.reference);
+      return named(type.reference());
```

Comment:
we resolve the reference to a reflection type in runtime instead of at startup to avoid issues with cyclical dependencies breaking things